x/interchainstaking/keeper/validators.go

```diff
@@ -92,3 +92,34 @@ func (k Keeper) IterateValidators(ctx sdk.Context, chainID string, fn func(index
 		i++
 	}
 }
+
+// GetValidatorByConsAddr returns validator by Consensus address.
+func (k Keeper) GetValidatorByConsAddr(ctx sdk.Context, chainID string, consAddr []byte) (types.Validator, bool) {
+	val := types.Validator{}
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.GetZoneValidatorsByConsAddrKey(chainID))
+	bz := store.Get(consAddr)
+	if len(bz) == 0 {
+		return val, false
+	}
+
+	k.cdc.MustUnmarshal(bz, &val)
+	return val, true
+}
+
+// SetValidatorByConsAddr set validator by Consensus address.
+func (k Keeper) SetValidatorByConsAddr(ctx sdk.Context, chainID string, val types.Validator) error {
```

Comment:
Do we want to store the val object here? Or simply the valoper address?

Storing the val object, means we have to update the validator in two places for any subsequent change (plus we are storing all the data twice). Just storing a reference to the valoper address is enough to lookup the canonical validator record however.

Reply:
Make sense, I will update it